pkg/awsclient/elb_helper.go

```diff
@@ -212,6 +215,123 @@ func (c *AwsClient) DeleteExternalLoadBalancer(extLoadBalancerArn string) error
 	return err
 }
 
+// CreateNetworkLoadBalancer should only return one new NLB at a time
+func (c *AwsClient) CreateNetworkLoadBalancer(lbName, scheme, subnet string) ([]LoadBalancerV2, error) {
+	i := &elbv2.CreateLoadBalancerInput{
+		Name:   aws.String(lbName),
+		Scheme: aws.String(scheme),
+		Subnets: []*string{
+			aws.String(subnet),
+		},
+		Type: aws.String("network"),
+	}
+
+	result, err := c.CreateLoadBalancerV2(i)
+	if err != nil {
+		return []LoadBalancerV2{}, err
+	}
+
+	// there should only be 1 NLB made, but since CreateLoadBalancerOutput takes in slice
+	// we return it as slice
+	loadBalancers := make([]LoadBalancerV2, 0)
+	for _, loadBalancer := range result.LoadBalancers {
+		loadBalancers = append(loadBalancers, LoadBalancerV2{
+			CanonicalHostedZoneNameID: aws.StringValue(loadBalancer.CanonicalHostedZoneId),
+			DNSName:                   aws.StringValue(loadBalancer.DNSName),
+			LoadBalancerArn:           aws.StringValue(loadBalancer.LoadBalancerArn),
+			LoadBalancerName:          aws.StringValue(loadBalancer.LoadBalancerName),
+			Scheme:                    aws.StringValue(loadBalancer.Scheme),
+			VpcID:                     aws.StringValue(loadBalancer.VpcId),
+		})
+	}
+	return loadBalancers, nil
+}
+
+// CreateExternalNLBTargetGroup creates the external NLB target group and returns the targetGroupArn
+func (c *AwsClient) CreateExternalNLBTargetGroup(nlbName, vpcID string) (string, error) {
+	i := &elbv2.CreateTargetGroupInput{
+		Name:                       aws.String(nlbName),
+		Port:                       aws.Int64(6443),
```

Comment:
Probably we should consider transforming Port and Protocols into constants, and use them here.

Reply:
noted. Since this is not being called in controller I will revisit.